sympy/core/exprtools.py

```diff
@@ -448,14 +448,12 @@ def as_expr(self):  # Factors
         args = []
         for factor, exp in self.factors.items():
             if exp != 1:
-                b, e = factor.as_base_exp()
                 if isinstance(exp, int):
```

Comment:
Could `exp` be a SymPy Integer?

Reply:
Yes and in fact that seems to be the norm (at least for the few simple manual tests that I ran). In that case the general case applies (`args.append(factor**exp)`) and still returns the right result. I must admit though that I'm wondering a bit about that special-casing of the `int` case. Is that only to have it sympified? I'm also unsure about the `_keep_coeff`.
Any further insight is much appreciated.

Reply:
`_keep_coeff` is just there to keep 2-arg muls from evaluating (see docstring)

Reply:
From a quick look at Factors it seems that all values should be Basic type so the check for `int` should, perhaps, just be `Integer`.

Reply:
What's strange though is that IIRC codecov marked the code for the `isinstance(exp, int)` case as being covered.
So both cases (int and Integer) should ideally end up using `_keep_coeff`?
I'll change it to `isinstance(exp, (int, Integer))` then.

Reply:
But the only place I can see an int potentially creeping in is at lines 360-361 where a default of 0 might be added to something that came from powers dict. If the raw 0 were changed to S.Zero then that would fix that (assuming that `as_powers_dict` might be returning an int. If the int case were fixed then problems like here will be avoided elsewhere.